### ISSUE TITLE
Using syntax/parse inside of a module+ submodule defined at phase 1 causes an internal error

### ISSUE THREAD

**Issue by lexi-lambda** [racket/racket#1492]
This is a pretty arcane combination of things, so I’m not sure exactly what’s going on here (and if it’s a macroexpander bug or a `syntax/parse` bug), but I thought I should report it. This program causes an error:

``` racket
#lang racket
(begin-for-syntax
  (module+ foo
    (require (for-syntax syntax/parse))
    (define-syntax (mac stx)
      (syntax-parse stx))))
```

``` racket
racket/collects/syntax/parse/private/parse.rkt:453:29: ?: internal error: fail-handler used out of context
  in: (fail-handler (failure pr es))
```

**Comment by AlexKnauth** [comment]
That error message looks like it comes from here:
https://github.com/racket/racket/blob/master/racket/collects/syntax/parse/private/runtime.rkt#L54-L56

So it's probably a `syntax/parse` bug and not a macroexpander bug.

**Comment by lexi-lambda** [comment]
It’s certainly possible, but it seems odd to me that something would work fine in an ordinary module but fail in a submodule, which is what leads me to wonder if it’s a macroexpander or syntax parameter issue.

**Comment by mflatt** [comment]
The rewritten macro expander doesn't raise an error for this program. Unless the bug is pressing, I'll file this under things to be fixed by replacing the expander in the near future.

**Comment by lexi-lambda** [comment]
Fixed since Racket v7.0.
